src/targeting/Semgrepignore.ml

```diff
@@ -150,7 +154,7 @@ let create ?(cli_patterns = []) ~default_semgrepignore_patterns
   *)
   let root_semgrepignore_exists =
     let root_dir = Ppath.to_fpath ~root:project_root Ppath.root in
-    let semgrepignore_path = Fpath.add_seg root_dir ".semgrepignore" in
+    let semgrepignore_path = Fpath.add_seg root_dir semgrepignore_filename in
```

Comment:
You should remove the tests where a path is passed, since this will not work.

Try this in utop: 

```ocaml
Fpath.add_seg (Fpath.v ".") "hello.txt";;
```

The above works. Now try: 

```ocaml
Fpath.add_seg (Fpath.v ".") "build/hello.txt" |> Fpath.to_string;;
```

The above fails.

So in fact only a filename can be passed.

I'm fine with this, in fact I prefer it. But if you want to propose a more flexible feature, let me know.